### PR TITLE
Mentors Dashboard to tsx

### LINKED ISSUE #1615: TypeScript: Mentors Dashboard
image

### PR DESCRIPTION
**Pull request by d-leeee**

### CODE REVIEW

File: src/components/admin/dashboards/Mentors.jsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 "use client";
-import { STATUSES, TAGS, COLUMNS, DROPDOWN } from "@/data/admin/Mentors.js";
+import { STATUSES, TAGS, COLUMNS, DROPDOWN } from "@/data/admin/Mentors.tsx";
```
Comment (shahdivyank) [inline src/components/admin/dashboards/Mentors.jsx]: import statements shouldn't need file extensions

File: src/components/form/Mentor.jsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { FIELDS, ATTRIBUTES } from "@/data/form/Mentors.js";
 import { useSession } from "next-auth/react";
 import { api } from "@/utils/api";
 import toaster from "@/utils/toaster";
-import { STATUSES } from "@/data/admin/Mentors.js";
+import { STATUSES } from "@/data/admin/Mentors.tsx";
```
Comment (shahdivyank) [inline src/components/form/Mentor.jsx]: import statements shouldn't need file extensions

File: src/data/admin/Mentors.tsx [hunk 3]
```diff
@@ -1,14 +1,60 @@
 import Checkbox from "@/components/Checkbox";
 import { AVAILABILITY } from "../form/Information";
 import { generateSelect, generateStatus } from "./Columns";
+import React from "react";
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: dont need this

File: src/data/admin/Mentors.tsx [hunk 4]
```diff
@@ -19,7 +65,7 @@ export const TAGS = [
   },
 ];
 
-export const COLUMNS = [
+export const COLUMNS: column[] = [
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: check tanstack table for the column defs: ColumnDef

File: src/data/admin/Mentors.tsx [hunk 5]
```diff
@@ -61,6 +107,7 @@ export const DROPDOWN = ({ object }) => {
               toggle={object.availability.includes(text)}
               text={text}
               key={index}
+              color
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: this should not be needed

Comment (d-leeee) [reply src/data/admin/Mentors.tsx]: `Property 'color' is missing in type '{ toggle: any; text: string; key: number; }' but required in type '{ toggle: any; onClick?: (() => void) | undefined; text?: string | undefined; color: any; `

File: src/data/admin/Mentors.tsx (hunk 4 above)
Comment (d-leeee) [reply src/data/admin/Mentors.tsx]: should i be changing the cell property or all the properties? Should i be importing tanstack?

File: src/data/admin/Mentors.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  discord: string;
+  status: number | string;
+}
+
+interface tagsDef {
+  text: string;
+  value: number;
+}
+
+interface columnsDef {
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: these definitions are provided by tanstack, use those instead

File: src/components/admin/dashboards/Mentors.jsx [hunk 7]
```diff
@@ -1,7 +1,5 @@
 "use client";
-import { TAGS, COLUMNS, DROPDOWN } from "@/data/admin/Mentors.js";
-import { STATUSES } from "@/data/Statuses";
-
+import { STATUSES, TAGS, COLUMNS, DROPDOWN } from "@/data/admin/Mentors";
```
Comment (shahdivyank) [inline src/components/admin/dashboards/Mentors.jsx]: make sure to pull from dev, there were changes to how statuses is being used

File: src/data/admin/Mentors.tsx [hunk 8]
```diff
@@ -2,8 +2,21 @@ import Checkbox from "@/components/Checkbox";
...
 
-export const TAGS = [
+type Person = {
+  name: string;
+  email: string;
+  discord: string;
+  status: string | number;
+};
+
+interface tagsDef {
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: use type

File: src/data/admin/Mentors.tsx [hunk 9]
```diff
@@ -14,39 +27,40 @@ export const TAGS = [
...
+    cell: (props: { getValue: () => string }) => <div>{props.getValue()}</div>,
   },
   generateStatus(STATUSES),
 ];
 
-export const DROPDOWN = ({ object }) => {
+export const DROPDOWN = ({
+  object,
+}: {
+  object: { availability: string[]; response: string };
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: extract into type

File: src/data/admin/Mentors.tsx [hunk 10]
```diff
@@ -14,39 +27,40 @@ export const TAGS = [
...
 ];
 
-export const COLUMNS = [
+export const COLUMNS: ColumnDef<Person, string>[] = [
   generateSelect(),
   {
     accessorKey: "name",
     header: "Name",
-    width: "w-3/12",
+    // width: "w-3/12",
```
Comment (shahdivyank) [inline src/data/admin/Mentors.tsx]: u should be able to extend the type definition

**Pull request closed without merging**